crates/miden-objects/src/note/note_tag.rs

```diff
@@ -154,18 +190,20 @@ impl NoteTag {
             return Err(NoteError::NoteTagUseCaseTooLarge(use_case_id));
         }
 
-        let execution_bits = match execution {
-            NoteExecutionMode::Local => PUBLIC_USECASE, // high bits set to `0b10`
-            NoteExecutionMode::Network => 0x40000000,   // high bits set to `0b01`
-        };
-
-        let use_case_bits = (use_case_id as u32) << 16;
-        let payload_bits = payload as u32;
-
-        Ok(Self(execution_bits | use_case_bits | payload_bits))
+        match execution {
+            NoteExecutionMode::Network => {
+                let use_case_bits = (NETWORK_PUBLIC_USECASE >> 16) as u16 | use_case_id;
+                Ok(Self::NetworkUseCase(use_case_bits, payload))
+            },
+            NoteExecutionMode::Local => {
+                // Convert directly to u32: type bits (0b11) + use_case_bits + payload
+                let tag_u32 = (0b10u32 << 30) | ((use_case_id as u32) << 16) | (payload as u32);
+                Ok(Self::LocalPublicAny(tag_u32))
```

Comment:
```suggestion
                let tag_u32 =
                    LOCAL_PUBLIC_ANY | ((use_case_id as u32) << 16) | (payload as u32);
                Ok(Self::LocalPublicAny(tag_u32))
```

After renaming `LOCAL_PUBLIC_USECASE` to `LOCAL_PUBLIC_ANY` we can just use the constant right? I would remove the comment since its outdated.